src/components/editor/faultTree/Editor.tsx

```diff
@@ -206,25 +206,31 @@ const Editor = () => {
 
   const redirectToInstance = (navigateFrom: string) => {
     if (faultTree?.manifestingEvent?.children) {
-      let map: { [name: string]: string } = {};
-      let children = faultTree?.manifestingEvent?.children;
-      if (faultTree?.manifestingEvent?.children.length > 0) {
-        faultTree?.manifestingEvent?.children.map((child) => {
-          if (child.children) {
-            children = [...children, ...child.children];
+      // Recursive function to traverse the tree and find the desired key-value pair
+      const findNodeValue = (children) => {
```

Comment:
@Kasmadei I mentioned at least twice that you should not implement a recursive traverse function but instead save the state in the FTA node. I even described it here:
https://github.com/kbss-cvut/fta-fmea-ui/issues/476#issuecomment-2195609473

So why you needed to implement recursive function here?

Reply:
Refactored. I found a way how to refactor this, so traverse function removed.